File: app/src/main/java/sb/yoon/kiosk/ItemListFragment.java
package sb.yoon.kiosk;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.LinearLayout;
import android.widget.ListView;
import android.widget.TableLayout;
import android.widget.TableRow;
import android.widget.TextView;
import android.widget.ViewFlipper;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.ListFragment;
import sb.yoon.kiosk.controller.KioskListAdapter;
import sb.yoon.kiosk.model.Menu;

import java.util.List;

public class ItemListFragment extends ListFragment {

    private List<Menu> menuList;

    public ItemListFragment(List<Menu> menuList) {
        this.menuList = menuList;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        KioskListActivity mainActivity = (KioskListActivity) getActivity();

        // 인덱스 표시 어댑터 설정
        KioskListAdapter adapter = new KioskListAdapter(menuList, mainActivity);
        // 어댑터를 설정
        setListAdapter(adapter);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        //View view = super.onCreateView(inflater, container, savedInstanceState);
        View view = inflater.inflate(R.layout.list_fragment, container, false);

        //
        /*ListView listView = view.findViewWithTag("testlist");
        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                System.out.println(parent);
            }
        });*/



        return view;


    }
}

File: app/src/main/java/sb/yoon/kiosk2/ItemListFragment.kt
package sb.yoon.kiosk2

import android.os.Bundle
import sb.yoon.kiosk2.controller.KioskListAdapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.ListFragment
import sb.yoon.kiosk2.model.Menu

class ItemListFragment(private val menuList: List<Menu>) : ListFragment() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val mainActivity = activity as KioskListActivity?

        // 인덱스 표시 어댑터 설정
        val adapter = KioskListAdapter(menuList, mainActivity)
        // 어댑터를 설정
        listAdapter = adapter
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.list_fragment, container, false)
    }
}